tests/Constraint/MultiConstraintTest.php

```diff
@@ -243,4 +243,64 @@ public function testMultipleMultiConstraintsMergingWithGaps()
         $this->assertEquals(new Bound('7.2.2.0-dev', true), $constraint->getLowerBound(), 'Expected lower bound does not match');
         $this->assertEquals(new Bound('8.0.0.0-dev', false), $constraint->getUpperBound(), 'Expected upper bound does not match');
     }
+
+    /**
+     * @dataProvider multiConstraintOptimizations
+     *
+     * @param string $constraints
+     */
+    public function testMultiConstraintOptimizations($constraints, ConstraintInterface $expectedConstraint)
+    {
+        // We're using the version parser here because that uses MultiConstraint::create() internally and
+        // thus tests our optimizations. It's just easier to write complex multi constraint instances
+        // using the string notation.
+        $parser = new VersionParser();
+        $this->assertSame((string) $expectedConstraint, (string) $parser->parseConstraints($constraints));
+    }
+
+    public function multiConstraintOptimizations()
+    {
+        return array(
+            'Test collapses contiguous' => array(
+                '^2.5 || ^3.0',
+                new MultiConstraint(array(
```

Comment:
array should be on newline here otherwise it looks like broken indent :)

Reply:
Done in 5c0356851773eb259c1557c09d146aa9e08d42ca.